Sync Settings updates for UI changes (#772)
obsidianmd/obsidian-help: en/Obsidian Sync/Select files and settings to sync.md

Any files or settings synced to your [[Local and remote vaults|remote vault]] contribute to your [[Sync limitations#How large can each remote vault be|storage space]]. By default, [[Introduction to Obsidian Sync|Obsidian Sync]] activates **selective sync** for the following file types:
- Images
- Audio
- Videos
- PDFs

To sync additional file types, manually toggle `Sync all other types`.

The default **vault configuration sync** settings include:
- Other file types
- Main settings
- Appearance
- Themes and snippets
- Hotkeys
- Active core plugin list
- Core plugin settings

> To sync community plugins, manually toggle `Active community plugin list` and `Installed community plugin list`.

Reviewer: I just looked and I think we always use bold for setting names, not code.

Author: ope, good catch.

Note that your [[Plans and storage limits|Sync plan]] defines the maximum file size you can sync. With the Standard plan you can sync files up to 5MB, while the Plus plan allows you to sync files up to 200MB.

> > [!info] Files previously synced will remain in your remote vault even if later excluded from sync. Configure the files and settings you wish to sync before initiating the process to avoid unnecessary use of storage space.

Reviewer: ```suggestion
> [!info] Adding a file to the **Excluded files** list will not remove the file from the remote vault if it has already been synced. Configure the files and settings you wish to sync before initiating the process to avoid unnecessary use of storage space.
```

Tried to improve the phrasing here a bit.

### Sync settings are not synced

> Sync settings themselves are not synced so that each each device can be separately configured according to your needs.

Reviewer: ```suggestion
Sync settings themselves are not synced so that each device can be separately configured according to your needs.
```

## Updating your synced vault settings

If you need to modify your sync settings across multiple devices, follow these steps:

> [!tip] In this context, the remote vault does not differentiate between primary and secondary devices; these terms are used for clarity.

### Primary device

The primary device is considered your source-of-truth. This is the device on which you make changes that you want to sync across all other devices.

1. Access **Settings** → **Sync**.
2. Activate the desired settings under **Vault configuration sync**.
3. Reload or Restart Obsidian. On mobile or tablet, this may require a force quit.
4. Allow time for the settings to synchronize with your remote vault.

### Secondary device(s)

The secondary device is any other device you use with Obsidian, such as your phone. It refers to devices that receive updates from the primary device.

1. Navigate to **Settings** → **Sync**.
2. Enable the desired settings under **Vault configuration sync**.
3. Wait for the changes to be downloaded from your remote vault.

> 4. Reload or Restart the application to ensure that your synchronized settings take effect. On mobile or tablet, this may require a force quit.

Reviewer: ```suggestion
4. Reload or restart the application to ensure that your synchronized settings take effect. On mobile or tablet, this may require a force quit.
```

## Hot reloading of settings

> [!question] Wondering what Hot Reloading is? Hot reloading refreshes only the modified files without resetting the app's state, but it does not necessarily apply those changes without a reload.

Obsidian has limited ability to update and apply configurations received from Sync. The following configurations will hot reload:

- Most Obsidian app configurations, including hotkeys and properties.
- Appearance configurations.
- Configurations for already enabled Obsidian Core plugins (e.g., the location of the daily notes template folder).

However, for CSS changes involving [[CSS snippets]] or [[Themes]], while the appearance configurations are received, the application must be reloaded to reflect visual changes.

The following configurations will not hot reload, and will require an application reload or restart:

- Graph view configurations.
- States of Obsidian Core plugins (e.g., whether Daily Notes is enabled).

> - Community plugin configurations.

Reviewer: I think we should remove this line. The Obsidian API provides a way for community plugins to reload when settings sync. Not all plugins are using this, but it's incorrect to say that plugin settings do not sync.

Author: To confirm, remove the community plugins line (92) or lines 89-92?

Reviewer: "Community plugin configurations" (just line 92)

Additionally, in rare cases, configurations that do not support hot reloading may be merged incorrectly or overwritten completely. For more details, see [[Troubleshoot Obsidian Sync#Conflict resolution|conflict resolution]].

> > [!abstract] Community Plugins and Hot-Reloading

Reviewer: ```suggestion
> [!abstract] Community plugins and hot-reloading
```

> > Community plugin developers can enable hot reloading for their configuration files using the [Plugin.onExternalSettingsChange() method](https://docs.obsidian.md/Reference/TypeScript+API/Plugin/onExternalSettingsChange). However, hot reloading may not be suitable for all plugins and could be disadvantageous in some scenarios.

Reviewer: I don't think this is an appropriate place to include details about the API. It isn't the right audience and the answer feels incomplete.

I think this information should be on the developer docs site, and if we want to link there, we should just say `"Are you a developer, learn how to [seamlessly integrate your plugin with Sync.](<link>)"`.

Author: I did want to call attention to it slightly, to get it more known thats it is a thing. The one liner you suggested better, I will commit an update.

### Sync settings are not synced

> Sync settings themselves are not synced so that each device can be separately configured according to your needs.

Reviewer: This is a pretty good tongue twister.

I wonder if we can avoid the reusing the word `sync` as the verb here to make this a bit less confusing.

Here's my attempt:

> Sync settings are not shared across devices. You must Each device must separately configure the settings on each device according to your needs.

Author: I think you may have missed a word?

Was it meany to be the below?
> Sync settings are not shared across devices. You must separately configure the settings on each device according to your needs.

Reviewer: oh oops, yea I just didn't delete some words in the edit. That's what I meant.

Author: > oh oops, yea I just didn't delete some words in the edit. That's what I meant.

Great, I'll push an update. You fine with me merging after this?
